### PR TITLE
Implement `ROW_GROUPS_PER_FILE` for Parquet

### PR DESCRIPTION
**Pull request by lnkuiper**
This PR implements the `ROW_GROUPS_PER_FILE` parameter for the Parquet writer. This parameter creates a new Parquet file if the current one has a specified number of row groups.

It is exact when only one thread is active, or if `PER_THREAD_OUTPUT` is true, and it is approximate if not.

For example:
```sql
SET threads=1;
CREATE TABLE bigdata AS SELECT i AS col_a, i AS col_b FROM range(0,10000) tbl(i);
COPY bigdata TO 'my_dir1' (FORMAT PARQUET, ROW_GROUP_SIZE 2000, ROW_GROUPS_PER_FILE 1);
-- 5 output files each containing 1 row group of ~2000 rows
COPY bigdata TO 'my_dir2' (FORMAT PARQUET, ROW_GROUP_SIZE 4000, ROW_GROUPS_PER_FILE 1);
-- 3 output files each containing 1 row group: 2 files with ~4000 rows, 1 file with the remaining ~2000
COPY bigdata TO 'my_dir3' (FORMAT PARQUET, ROW_GROUP_SIZE 2000, ROW_GROUPS_PER_FILE 2);
-- 3 output files: 2 files with 2 row groups of ~2000 rows, and 1 file with 1 row group of ~2000 rows
```

If multiple threads are active, the number of row groups in a file may slightly exceed the specified number of row groups, as we want to write to the file without too much locking, exactly like `FILE_SIZE_BYTES`. However, if `PER_THREAD_OUTPUT` is set, only one thread writes to each file, and it becomes accurate again.

### CODE REVIEW

**Comment by lnkuiper** [comment]
I think the failures are spurious

File: src/include/duckdb/function/copy_function.hpp [hunk 1]
```diff
@@ -107,6 +107,10 @@ typedef idx_t (*copy_desired_batch_size_t)(ClientContext &context, FunctionData
 
 typedef idx_t (*copy_file_size_bytes_t)(GlobalFunctionData &gstate);
 
+typedef bool (*copy_rotate_files_t)(FunctionData &bind_data);
+
+typedef bool (*copy_rotate_next_file_t)(GlobalFunctionData &gstate, FunctionData &bind_data);
```
Comment (Mytherin) [inline src/include/duckdb/function/copy_function.hpp:112]: Maybe `copy_file_size_bytes` can be superseded and make use of this callback as well?

**Comment by lnkuiper** [comment]
I think this PR is ready to go!

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**